core/drivers/scmi-msg/smt.c

```diff
@@ -91,7 +91,8 @@ static struct smt_header *channel_to_smt_hdr(struct scmi_msg_channel *chan)
  * references to input payload in secure memory and output message buffer
  * in shared memory.
  */
-static void scmi_proccess_smt(unsigned int channel_id, uint32_t *payload_buf)
+static void __maybe_unused scmi_process_smt(unsigned int channel_id,
```

Comment:
Why is `__maybe_unused` needed? Does it makes sense to have `CFG_SCMI_MSG_SMT_FASTCALL_ENTRY`, `CFG_SCMI_MSG_SMT_FASTCALL_ENTRY` and `CFG_SCMI_MSG_SMT_THREAD_ENTRY` all disabled at the same time while still having `CFG_SCMI_MSG_SMT=y`? (this file is not built when `CFG_SCMI_MSG_SMT != y`).

Reply:
Likely not, but I found it simpler stubbing each based on their relative CFG_SCMI_MSG_xxx_ENTRY and have the generic function left with this `__maybe_unused`. I think I can remove this attribute if preferred.

Reply:
Yes, `__maybe_unused` is wrong since it cannot happen that the function is unused (at least one `CFG_` is expected to be set).